django/db/views.py

```diff
@@ -1635,7 +1644,9 @@ def __init__(self,dataset,is_authenticated=False):
             self.has_otherreagents_for_datarecords(self.dataset_id)
         self.small_molecule_queryset = \
             self.small_molecules_for_dataset(self.dataset_id)
-                
+        self.has_small_molecules_for_datarecords = \
```

Comment:
valid point. follows is an explanation of my thought process (not an argument against your point):
- this is an incremental change, so I decided to copy the pattern from the 4 other instances of this pattern.  
- If pressed to divine my original intentions here, I would guess that I'd originally constructed DatasetManager as a static utility, with no state (i.e. dataset_id) but that for some reason I converted it to a stateful object. 
- considering all of this, I've left the pattern in place.  Argument being that the repeating of the pattern makes it more clear that this is the latest of a series.
- in the past, we have discussed that whole-scale revisions should be separated from feature implementations, (e.g. reformatting vs. adding a feature).  Ideally, every new feature commit should be preceded or followed by a fix of all incorrect code segments that it touches.  (should the fix be before or after the feature commit?  before, presumably, so that we don't have to have this discussion :-) ).
- If redoing this file I would do it differently.  (There are arguments for whole scale rework of this, and other parts).
- if we are reviewing and fixing this codebase, I would argue that doing it piecemeal in this fashion is ineffective.
- that said, your point is valid.  Shall this pattern be fixed for all instances then?